packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -566,37 +574,47 @@ describe('KeyringController', () => {
         });
 
         describe('when there is an existing vault', () => {
-          it('should return existing vault', async () => {
+          it('should not create a new vault or keychain', async () => {
             await withController(
               { cacheEncryptionKey },
               async ({ controller, initialState }) => {
                 const initialSeedWord = await controller.exportSeedPhrase(
                   password,
                 );
+                expect(initialSeedWord).toBeDefined();
                 const initialVault = controller.state.vault;
+
                 await controller.createNewVaultAndKeychain(password);
+
                 const currentSeedWord = await controller.exportSeedPhrase(
                   password,
                 );
-                expect(initialSeedWord).toBeDefined();
                 expect(initialState).toStrictEqual(controller.state);
-                expect(currentSeedWord).toBeDefined();
                 expect(initialSeedWord).toBe(currentSeedWord);
                 expect(initialVault).toStrictEqual(controller.state.vault);
               },
             );
           });
-        });
 
-        cacheEncryptionKey &&
-          it('should set encryptionKey and encryptionSalt in state', async () => {
-            // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            withController({ cacheEncryptionKey }, async ({ initialState }) => {
-              expect(initialState.encryptionKey).toBeDefined();
-              expect(initialState.encryptionSalt).toBeDefined();
+          cacheEncryptionKey &&
+            // TODO: Re-enable this after we properly clear the key and salt from state upon lock
+            // eslint-disable-next-line jest/no-disabled-tests
+            it.skip('should set encryptionKey and encryptionSalt in state', async () => {
+              await withController(
+                { cacheEncryptionKey },
+                async ({ controller }) => {
+                  await controller.setLocked();
+                  expect(controller.state.encryptionKey).toBeUndefined();
```

Comment:
The test currently fails on this line, but this will be fixed in #4514, at which point we can unskip this test.

There is no way to test this using just public API (that I could think of) until we merge #4514